### PR TITLE
*: always share block cache

### LINKED ISSUE #12936: Remove the suppport for specifying block cache for cf manually
## Feature Request

For historical reason, we support specifying block cache for each cf for either raft db or kv db. But since we support sharing cache between CFs and dbs (3 years ago), these settings are not practical anymore. They are hard to use and easy to make mistake. We better remove those settings and always enabling share cache for simplicity.

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Close #12936

What's Changed:

There has been 5 months and nobody objects it so let's always enable share block cache.

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
    https://github.com/pingcap/docs-cn/pull/12241

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Breaking backward compatibility
    Use separate cache for each CF is not supported anymore.

### Release note 

```release-note
remove the support of specifying separate cache for each CF
```

### CODE REVIEW

File: src/storage/config.rs [hunk 1]
```diff
@@ -193,8 +193,6 @@ impl Default for FlowControlConfig {
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct BlockCacheConfig {
-    #[online_config(skip)]
-    pub shared: bool,
```
Comment (tabokie) [inline src/storage/config.rs:197]: Option<bool> for compatibility?

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

**Review by tabokie — APPROVED** [review]

**Pull request merged**